### PR TITLE
Support molmo in turbomind

### PR DESCRIPTION
**Pull request by lvhan028**
model: allenai/Molmo-7B-D-0924

### CODE REVIEW

**Comment by irexyc** [comment]
It seems doesn't support text only input like
```
pipe('who are you?')

# or
messages = [
    dict(role='user', content=[
        dict(type='text', text='你是谁')
    ])
]
pipe(messages)
```

File: lmdeploy/vl/templates.py [hunk 1]
```diff
@@ -428,6 +428,73 @@ class GLM4VChatTemplateWrapper(VLChatTemplateWrapper):
...
+                else:
+                    logger.error(f'unexpected content type {message}')
+            out_messages.append(message)
+
+        await asyncio.gather(*[
+            asyncio.get_event_loop().run_in_executor(None, _inner_call, i,
+                                                     messages, out_messages)
+            for i in range(len(messages))
+        ])
+        #
```
Comment (irexyc) [inline lmdeploy/vl/templates.py:483]: Concurrency cannot be used here or the input of `[user, assistant, user]` may become `[assistant, user, user]`

Comment (irexyc) [reply lmdeploy/vl/templates.py:483]: And after fixing the order, I find the position of `Assistant` seems not right. I decode the input token and the result is 
```
" User: Describe the two images in detail. Assistant:Assistant: xxx...User: What are the similarities and differences between these two images. "  
```
(use the multi-round example [here](https://github.com/InternLM/lmdeploy/blob/main/docs/en/multi_modal/qwen2_vl.md))


If the input only has one user content and decode the input ids, the result is 
```
" User: Describe the two images in detail. Assistant:"
```

**Comment by lvhan028** [comment]
> It seems doesn't support text only input like
> 
> ```
> pipe('who are you?')
> 
> # or
> messages = [
>     dict(role='user', content=[
>         dict(type='text', text='你是谁')
>     ])
> ]
> pipe(messages)
> ```

Fixed

File: lmdeploy/vl/templates.py (hunk 1 above)
Comment (lvhan028) [reply lmdeploy/vl/templates.py:483]: Fixed

**Comment by irexyc** [comment]
When I run the `multi-image multi-round conversation` [example](https://github.com/InternLM/lmdeploy/blob/main/docs/en/multi_modal/qwen2_vl.md)

```
  File "/home/chenxin/ws3/topk/lmdeploy/vl/model/molmo.py", line 99, in forward
    inputs = self.processor.process(images=message['images'],
  File "/home/chenxin/.cache/huggingface/modules/transformers_modules/preprocessing_molmo.py", line 127, in process
    tokens = self.get_tokens_input(
  File "/home/chenxin/.cache/huggingface/modules/transformers_modules/preprocessing_molmo.py", line 104, in get_tokens_input
    prompt = "User: " + prompt + " Assistant:"
TypeError: can only concatenate str (not "NoneType") to str
```

**Review by irexyc — APPROVED** [review]

**Review by lzhangzz — APPROVED** [review]

**Pull request merged**